src/app/child-dev-project/children/children-list/recent-attendance-blocks/recent-attendance-blocks.component.ts

```diff
@@ -58,27 +56,30 @@ export class RecentAttendanceBlocksComponent implements OnInitDynamicComponent {
     });
   }
 
-  onInitFromDynamicConfig(config: ColumnCellConfig) {
-    this.filterByInstitution = config.config.filterByInstitution;
+  async onInitFromDynamicConfig(config: ColumnCellConfig) {
+    this.filterByActivityType = config.config.filterByActivityType;
+
     if (config.hasOwnProperty("entity")) {
       this.child = config.entity as Child;
-      this.childrenService
-        .getAttendancesOfChild(this.child.getId())
-        .subscribe((result) => this.prepareAttendanceData(result));
-    }
-  }
 
-  private prepareAttendanceData(loadedEntities: AttendanceMonth[]) {
-    this.attendanceList = loadedEntities.sort((a, b) => {
-      // descending by date
-      if (a.month > b.month) {
-        return -1;
+      let activities = await this.attendanceService.getActivitiesForChild(
+        this.child.getId()
+      );
+      if (this.filterByActivityType) {
+        activities = activities.filter(
+          (a) => a.type.id === this.filterByActivityType
+        );
       }
-      if (a.month < b.month) {
-        return 1;
+
+      this.attendanceList = [];
+      for (const activity of activities) {
+        const attendanceRecord = await this.attendanceService.getActivityAttendanceForPeriod(
```

Comment:
Should be also fixed with index. Currently, for each activity all notes are loaded and filtered. Would be more efficient to load all notes and distribute them to the correct attendance.